src/Core.Nhibernate/Models/AuthorizationCode.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace IdentityServer3.Contrib.Nhibernate.Models
```

Comment:
Where do these models come from? Why are they necessary?

Reply:
These have been created to allow the process previously being done by custom Json serialisers to be replaced with an Automapper object-object equivalent. The serialisers were writing directly to the database and performing object structure conversions in the process. This enables separation of object mapping, serialisation and IO operations properly.